rust/automerge/src/query/list_state.rs

```diff
@@ -0,0 +1,142 @@
+use crate::op_tree::{LastInsert, OpTreeNode};
+use crate::query::QueryResult;
+use crate::types::{Key, ListEncoding, Op};
+
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct ListState {
+    pub(crate) encoding: ListEncoding,
```

Comment:
Can we make most of these fields private and put the ones which are needed to be read crate wide (`last_seen`, `target`, `index`, and `pos`) behind accessors?